tests/ui/proc-macro/signature-proc-macro-attribute.stderr

```diff
@@ -4,8 +4,8 @@ error: attribute proc macro has incorrect signature
 LL | pub fn bad_input(input: String) -> TokenStream {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ incorrect number of function parameters
    |
-   = note: expected signature `fn(proc_macro::TokenStream, proc_macro::TokenStream) -> proc_macro::TokenStream`
-              found signature `fn(std::string::String) -> proc_macro::TokenStream`
+   = note: expected signature `fn(proc_macro::TokenStream, proc_macro::TokenStream) -> TokenStream`
+              found signature `fn(std::string::String) -> TokenStream`
```

Comment:
Discussion: this diff is kinda interesting because on a brief look, it reads to me almost as if `proc_macro::TokenStream` is a distinct type from `TokenStream`, even though they are the exact same definition.

Reply:
Yeah, I noticed this too. I'm not sure what the best approach for this might be. In this case in particular, the return types should both be `_`. I'm also thinking that maybe what we should actually be doing is looking if the type names are the same in both cases, and *if so*, print the full path, which is kind of the opposite of what this PR does.

Reply:
Yeah I understand this gets quickly very complicated (and probably outright unfeasible because these diagnostics are not emitted in the same place, and maybe not even on the same IR, lol).

Reply:
For now, I'll close this PR and leave it for future reference.